doma-processor/src/main/java/org/seasar/doma/internal/apt/annot/Annotations.java

```diff
@@ -191,7 +193,21 @@ private ScopeClass newScopeClass(TypeMirror scopeType) {
     TypeElement typeElement = ctx.getMoreTypes().toTypeElement(scopeType);
     List<? extends Element> members = ctx.getMoreElements().getAllMembers(typeElement);
     List<ExecutableElement> methods = new ArrayList<>(ElementFilter.methodsIn(members));
-    return new ScopeClass(type, methods);
+    methods.removeIf(m -> m.getAnnotation(ScopeMethod.class) == null);
+
+    List<TypeParameterDeclaration> typeParameterDeclarations = type.getTypeParameterDeclarations();
+    Map<TypeMirror, TypeMirror> typeParameterMap = new HashMap<>();
```

Comment:
It is better to remove `typeParameterMap` introducing a new class `ScopeMethodAdapter` as follows:

```java
List<ScopeMethodAdapter> methodAdapters = new ArrayList<>(method.size());
for (ExecutableElement method : methods) {
  List<TypeMirror> resolvedParameters = ...
  ScopeMethodAdapter methodAdapter = new ScopeMethodAdapter(method, resolvedParameters);
  methodAdapters.add(methodAdapter);
}
return new ScopeClass(scopeType, methodAdapters);
```

`ScopeMethodAdapter` represents the method which will be generated in an EntityMetamodel.